src/api/fleet_addon_config.rs

```diff
@@ -239,6 +244,55 @@ pub struct FeatureGates {
 
     /// Enables experimental Helm operations support.
     pub experimental_helm_ops: bool,
+
+    // Enables syncing of feature gates to a ConfigMap.
+    pub config_map: Option<FeaturesConfigMap>,
+}
+
+impl FeatureGates {
+    pub(crate) fn has_config_map_ref(&self) -> bool {
+        if let Some(_reference) = self.config_map.as_ref().and_then(|c | c.reference.as_ref()) {
+            return true;
+        }
+        return false;
+    }
+
+    pub(crate) fn merge_features(&self, input: &String) -> String {
+        let prefix = String::from(" |-\n");
+        let stripped_input = input.strip_prefix(&prefix).unwrap_or(&input);
+        let mut values: FleetChartValues = serde_yaml::from_str(stripped_input).unwrap_or_default();
+        let mut extra_env_map: BTreeMap<String, String> = BTreeMap::new();
+
+        // If there are existing extraEnv entries, convert them to a map.
+        if let Some(extra_env) = values.extra_env {
+            extra_env_map = extra_env.into_iter().map(|var| (var.name.unwrap_or_default(), var.value.unwrap_or_default())).collect();
+        }
+
+        // Sync the feature flags to the map.
+        extra_env_map.insert(String::from(EXPERIMENTAL_HELM_OPS), self.experimental_helm_ops.to_string());
+        extra_env_map.insert(String::from(EXPERIMENTAL_OCI_STORAGE), self.experimental_oci_storage.to_string());
+
+        // Convert the map back to list.
+        values.extra_env = Some(extra_env_map.iter()
+            .map(|(key, value)|
+            EnvironmentVariable{name: Some(key.clone()), value: Some(value.clone())})
+            .collect());
+
+        //Return the serialized updated values.
+        return prefix + &serde_yaml::to_string(&values).unwrap();
```

Comment:
I would suggest to do something with this unwrap statement. We probably need to surface this error and backoff it against reconcile error handling, otherwise we might end up panicking here and lead the pod into CrashLoopBackOff state.